src/base/io/Async.cpp

```diff
@@ -0,0 +1,78 @@
+#include "base/io/Async.h"
+
+
+#if defined(XMRIG_UV_PERFORMANCE_BUG)
+#include <sys/eventfd.h>
+#include <sys/poll.h>
+#include <unistd.h>
+#include <cstdlib>
+
+
+namespace xmrig {
+
+
+uv_async_t::~uv_async_t()
+{
+    close(m_fd);
+}
+
+
+static void on_schedule(uv_poll_t *handle, int status, int events)
+{
+    static uint64_t val;
+    uv_async_t *async = reinterpret_cast<uv_async_t *>(handle);
+    for (;;) {
+        int r = read(async->m_fd, &val, sizeof(val));
+
+        if (r == sizeof(val))
```

Comment:
I don't understand this part. If it successfully reads 8 bytes (uint64_t), it just keeps reading it again. How is async called then?

Reply:
Ok, I figured it out. It reads all incoming data and then fails with EAGAIN, triggering async call.